lib/omniauth/strategies/tiktok_oauth2.rb

```diff
@@ -1,7 +1,7 @@
 require 'oauth2'
 require 'omniauth/strategies/oauth2'
 
-OAuth2::Response.register_parser(:tiktok, ['application/json']) do |body|
+OAuth2::Response.register_parser(:tiktok, []) do |body|
   JSON.parse(body)['data']
```

Comment:
Looks like you should return the unparsed body as error handling.

```suggestion
  JSON.parse(body).fetch('data') rescue body
```